src/api/__tests__/Item.test.js

```diff
@@ -359,6 +371,97 @@ describe('api/Item', () => {
                 expect(item.errorCallback).toBe(noop);
             });
         });
+
+        describe('with a cached item', () => {
+            beforeEach(() => {
+                item.getCache = () => ({
+                    get: jest.fn().mockReturnValue('success'),
+                    has: jest.fn().mockReturnValue(true),
+                    merge: jest.fn(),
+                    set: jest.fn(),
+                });
+                item.shareSuccessHandler = jest.fn();
+                item.errorHandler = jest.fn();
+                item.getUrl = jest.fn().mockReturnValue('url');
+                item.xhr = {
+                    put: jest.fn().mockReturnValueOnce(Promise.resolve('success')),
+                };
+            });
+
+            afterEach(() => {
+                jest.resetModules();
+                jest.restoreAllMocks();
+            });
+
+            test('should skip the xhr if the item is cached', async () => {
+                const successCallback = jest.fn();
+                const errorCallback = jest.fn();
+                await item.share(file, 'access', successCallback, errorCallback);
+                expect(successCallback).toHaveBeenCalledWith('success');
+                expect(errorCallback).not.toHaveBeenCalled();
+                expect(item.errorHandler).not.toHaveBeenCalled();
+                expect(item.shareSuccessHandler).not.toHaveBeenCalled();
+                expect(item.xhr.put).not.toHaveBeenCalled();
+                expect(item.getUrl).not.toHaveBeenCalled();
+                expect(item.successCallback).toBeUndefined();
```

Comment:
Ideally, unit tests should focus on testing the expected input/output for a given method call. Do we need to test these internal details?